python/ray/data/datasource/file_based_datasource.py

```diff
@@ -65,32 +67,58 @@ class FileShuffleConfig:
     """Configuration for file shuffling.
 
     This configuration object controls how files are shuffled while reading file-based
-    datasets.
+    datasets. The random seed behavior is determined by the combination of ``seed``
+    and ``reseed_after_epoch``:
+
+    - If ``seed`` is None, the random seed is always None (non-deterministic shuffling).
+    - If ``seed`` is not None and ``reseed_after_epoch`` is False, the random seed is
+      constantly ``seed`` across epochs.
+    - If ``seed`` is not None and ``reseed_after_epoch`` is True, the random seed is
```

Comment:
We don't need to mention this case, `reseed_after_epoch` is only relevant when seed is set.